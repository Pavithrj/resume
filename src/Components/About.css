.aboutcontaner {
    margin-left: 10px;
}