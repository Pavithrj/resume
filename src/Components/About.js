import { useContext } from 'react';
import './About.css';
import ProfileContext from './ProfileContext';

function About() {
    const profileData = useContext(ProfileContext);
    const { details } = profileData.about;
    return (
        <div className='aboutcontaner'>
            <h3>About</h3>
            <div>{details}</div>
        </div>
    )
}

export default About;