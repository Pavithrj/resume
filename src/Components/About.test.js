import { render } from "@testing-library/react";
import ProfileContext from './ProfileContext';
import profileData from '../Details.json';
import About from "./About";

describe('Contact Component', () => {
    const customRender = (ui, { providerProps, ...renderOptions }) => {
        return render(
            <ProfileContext.Provider value={profileData}>{ui}</ProfileContext.Provider>,
            renderOptions,
        )
    }
    it('should print contact details', () => {
        const { asFragment } = customRender(<About />, {
            value: profileData
        });
        expect(asFragment()).toMatchSnapshot();
    })
})