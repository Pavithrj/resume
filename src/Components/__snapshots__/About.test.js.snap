// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should print contact details 1`] = `
<DocumentFragment>
  <div
    class="aboutcontaner"
  >
    <h3>
      About
    </h3>
    <div>
      As a Principal Software Engineer with over 9.2 years of experience in Front End Development, I have a proven track record of designing and implementing high-quality, responsive web applications using React, Redux, and Typescript/ JavaScript. My expertise in developing fast and responsive web designs has allowed me to create userfriendly interfaces that improve user engagement and satisfaction. In my most recent role as a Principal Software Engineer, I have played a key role in developing and implementing front-end architecture, optimizing application performance, and mentoring junior developers.
    </div>
  </div>
</DocumentFragment>
`;
